src/application-navigation/ApplicationNavigation.jsx

```diff
@@ -0,0 +1,192 @@
+   */
+  notifications: PropTypes.object, // eslint-disable-line react/no-unused-prop-types
+  /**
+   * By default, the ApplicationNavigation component will resolve any rendered NavigationPrompts prior to
+   * communicating navigation item selections with `onSelectNavigationItem`. If `disablePromptsForNavigationItems`
+   * is provided, no NavigationPrompts are resolved when navigation items are selected.
+   */
+  disablePromptsForNavigationItems: PropTypes.bool,
+  /**
+   * By default, the ApplicationNavigation component will resolve any rendered NavigationPrompts prior to
+   * communicating logout selection with `onSelectLogout`. If `disablePromptsForLogout` is provided,
+   * no NavigationPrompts are resolved when logout is selected.
+   */
+  disablePromptsForLogout: PropTypes.bool,
+  /**
+   * The Object (or function that returns an Object) that specifies the messages
+   * used to prompt the user when navigation items are selected while NavigationPrompts
+   * are rendered by the ApplicationNavigation content.
+   */
+  navigationPromptResolutionOptions: navigationPromptResolutionOptionsShape,
+};
+
+const ApplicationNavigation = ({
+  children,
+  notifications,
+  titleConfig,
+  extensionItems,
+  onSelectExtensionItem,
+  hero,
+  navigationItems,
+  activeNavigationItemKey,
+  onSelectNavigationItem: propOnSelectNavigationItem,
+  userConfig,
+  onSelectSettings,
+  onSelectHelp,
+  onSelectLogout: propOnSelectLogout,
+  utilityItems,
+  onSelectUtilityItem,
+  onDrawerMenuStateChange,
+  disablePromptsForNavigationItems,
+  disablePromptsForLogout,
+  navigationPromptResolutionOptions,
+}) => {
+  const navigationPromptCheckpointRef = useRef();
+
+  const onSelectNavigationItem = useCallback((selectedItemKey) => {
+    if (disablePromptsForNavigationItems) {
+      propOnSelectNavigationItem(selectedItemKey);
+      return;
+    }
+
+    navigationPromptCheckpointRef.current.resolvePrompts(navigationPromptResolutionOptions).then(() => {
+      propOnSelectNavigationItem(selectedItemKey);
+    }).catch((e) => { if (e) throw e; });
+  }, [disablePromptsForNavigationItems, navigationPromptResolutionOptions, propOnSelectNavigationItem]);
+
+  const onSelectLogout = useCallback(() => {
+    if (disablePromptsForLogout) {
+      propOnSelectLogout();
+      return;
+    }
+
+    navigationPromptCheckpointRef.current.resolvePrompts(navigationPromptResolutionOptions).then(() => {
+      propOnSelectLogout();
+    }).catch((e) => { if (e) throw e; });
+  }, [disablePromptsForLogout, navigationPromptResolutionOptions, propOnSelectLogout]);
+
+  return (
+    <TerraApplicationNavigation
+      hero={hero}
+      notifications={notifications}
+      titleConfig={titleConfig}
+      navigationItems={navigationItems}
+      onSelectNavigationItem={propOnSelectNavigationItem && onSelectNavigationItem}
+      activeNavigationItemKey={activeNavigationItemKey}
```

Comment:
Do consumer have the ability to by-pass the navigation prompting by supplying a new `activeNavigationItemKey`?

Reply:
They do, and we can't really prevent them from doing so. They would have to render their own NavigationPromptCheckpoint around ApplicationNavigation if there was an alternative way to change the selected navigation items.

Reply:
Can we make a note of it in the description or docs that they will be by-passing the navigation promptings?